File: hueservice/models/room.go
package models

type Room struct {
	Name      string     `json:"name,omitempty"`
	Lights    []string   `json:"lights,omitempty"`
	Type      string     `json:"type,omitempty"`
	Scenes    []*Scene   `json:"scenes,omitempty"`
	ID        int        `json:"-"`
	RoomState *RoomState `json:"roomState"`
}

File: webpanel/src/app/common/domain/lights/room.ts
export interface LightRoom {
  name: string;
  lights: number[];
  roomState: RoomState;
  scenes: Scene[];
}

export interface Scene {
  name: string;
  group: string;
}

export interface RoomState {
  on: boolean;
  brightness: number;
}

export interface GroupModification {
  groupName: string;
  on: boolean;
  sceneName: string;
  bri: number;
}
